core/runtime/types.hpp

```diff
@@ -41,6 +42,17 @@ namespace kagome::runtime {
    * space
    */
   using WasmOffset = uint32_t;
+
+  /**
+   * Splits 64 bit wasm span on 32 bit pointer and 32 bit address
+   */
+  static constexpr std::pair<WasmPointer, WasmSize> splitSpan(WasmSpan span) {
+    auto unsigned_result = static_cast<uint64_t>(span);
+    uint64_t minor_part = unsigned_result & 0xFFFFFFFFLLU;
+    uint64_t major_part = (unsigned_result >> 32u) & 0xFFFFFFFFLLU;
+
+    return {minor_part, major_part};
+  }
```

Comment:
Try to use another, but more elegant way. Look at [here](https://blog.tartanllama.xyz/structured-bindings/)

Reply:
Oh, I see. But as these types are just aliases to built-in integer types, I'm not sure it's gonna work here, but it could work for WasmResult.